backend/src/main/java/com/htecgroup/skynest/controller/UserController.java

```diff
@@ -204,7 +204,7 @@ public ResponseEntity<UserResponse> getUser(@PathVariable UUID uuid) {
   @PutMapping("/{uuid}")
   public ResponseEntity<UserResponse> editUser(
       @Valid @RequestBody UserEditRequest userEditRequest, @PathVariable UUID uuid) {
-    userService.authorizeAccessToUserDetailsWith(uuid);
+    userService.authorizeEditUserDetailsWith(uuid);
```

Comment:
Discussed internally how this can be achieved with custom annotation. @DarkoVeljanoskiHtec pls sync with Marko and apply the same changes in your PR